pkg/controller/machine_safety_test.go

```diff
@@ -71,6 +71,121 @@ var _ = Describe("machine", func() {
 		Entry("one machineset", nil, map[string]string{"machineset-1": ""}, nil, []string{"machineset-1"}, nil),
 	)
 
+	DescribeTable("##unfreezeMachineSetsAndDeployments",
+		func(machineSetExists, machineSetIsFrozen, parentExists, parentIsFrozen bool) {
+			createController := func(stop <-chan struct{}, objects ...runtime.Object) *controller {
+				fakeUntypedClient := fakeuntyped.NewSimpleClientset(objects...)
+				fakeTypedClient := &faketyped.FakeMachineV1alpha1{
+					&fakeUntypedClient.Fake,
+				}
+
+				controlMachineInformerFactory := machineinformers.NewSharedInformerFactory(fakeUntypedClient, 100*time.Millisecond)
+				defer controlMachineInformerFactory.Start(stop)
+
+				machineSharedInformers := controlMachineInformerFactory.Machine().V1alpha1()
+				machineSets := machineSharedInformers.MachineSets()
+				machineDeployments := machineSharedInformers.MachineDeployments()
+
+				return &controller{
+					controlMachineClient:    fakeTypedClient,
+					machineSetLister:        machineSets.Lister(),
+					machineDeploymentLister: machineDeployments.Lister(),
+					machineSetSynced:        machineSets.Informer().HasSynced,
+					machineDeploymentSynced: machineDeployments.Informer().HasSynced,
+				}
+			}
+
+			testMachineSet := &machinev1.MachineSet{
```

Comment:
It might be better to enhance the existing newMachineSet function to handle this.

I will do this. You can rebase after that, perhaps?

Reply:
Thought it will be too generic, but will wait for your changes and then adapt accordingly

Reply:
@PadmaB I have committed the change https://github.com/amshuman-kr/machine-controller-manager/commit/bf1ccb4a4cdd89ea8896457c4aeabda7a216503c to complete the helper function to initialise objects.

But I will create the PR only tomorrow after I have added the full round-trip test.